\chapter{Data Structure: Fluorescence Intensity Profiles}

Nanocoding \cite{Jo_etal_2007_PNAS}, \cite{Jo_etal_2009}, invented in LMCG, is a more advanced DNA barcoding technique, to obtain genome-wide restriction maps. In nanocoding, molecules are labeled with the restriction enzyme Nb.BbvCI, which cleaves the sequence GC\^\ TGAGG on single strands of double-stranded DNA molecules. The cleaved sites are made detectable by nick translation using fluorochrome-labelled nucleotides. The cleaved sites are called punctates. The labeling is done in test tubes and then presented on nanoscale tubes (fig \ref{fig:Fig2_NanoSlit}) for imaging and subsequent measurements. 
\begin{wrapfigure}{l}{0.5\textwidth}
%\includegraphics[width=0.9\linewidth]{Images/Image_Nanoslit.jpg} 
\begin{center}
\includegraphics[width=0.4\textwidth, bb=0 0 700 350]{Images/Image_Nanoslit.pdf}
\end{center}
\caption{Nanoslits where DNA molecules are presented}
\label{fig:Fig2_NanoSlit}
\end{wrapfigure}
The nanoslits are 1000 nm wide, 100 nm deep and 10 $\mu$m apart. DNA molecules are inherently negatively charged. Hence, they electrokinetically move through the microchannels as relaxed coils until approaching a nanoslit entrance. Then, as one of a molecule enters the nanoslit, it transiently elongates. This is aided by the low-ionic-strength buffers containing the DNA molecules. The DNA molecules are then deposited onto positively charged glass surface. The surface with DNA are then stained with YOYO-I dye, so they can be imaged by fluorescence microscopy. An argon ion laser-illuminated inverted Zeiss 135M microscope was used to image the DNA molecules. An integrated image acquisition and machine vision system was developed in LMCG to automatically detect and analyze molecule within the collected image data. Details of the image processing steps can be found in \cite{Dimalanta_etal_2004_AnalChem}, \cite{Jo_etal_2007_PNAS}, \cite{Ravindran_Gupta_2015_GigaScience}. 

The image processing softwares scans through images of surfaces presented with DNA molecules, as shown below in fig \ref{fig:Fig2_FrameImage}. 
\begin{figure}[h!]
\begin{center}
%\includegraphics[width=0.55\textwidth, bb=0 0 700 350]{Images/Image4_FrameImage.png}
\includegraphics[scale = 0.45, bb=0 0 1000 600]{Images/Image4_FrameImage.pdf}
\end{center}
\caption{Image of a surface with stained DNA molecules}
\label{fig:Fig2_FrameImage}
\end{figure}



The biggest advantage over optical mapping is being able to retain long DNA molecules intact and hence improving labeling efficiency. DNA molecules are stained with an intercalating dye YOYO-1. It is a green fluorescent dye which belongs to the family of monomethine cyanine dyes and is a tetracationic homodimer of Oxazole Yellow (abbreviated YO, hence the name YOYO). The labelled molecules are imaged by microscopes equipped with two CCD cameras, that have two filtering optics for green and red colors. The green channel acquires images of DNA backbone stained with YOYO-1, and the red channel acquires images of sequence-specific decorations of Alexa Fluor 547 punctuates via fluorescence resonance energy transfer (7). The restriction maps are obtained by meticulous image processing software that analyze the images from the green and red channels, to measure the lengths between two consecutive punctates.  \\



Show examples of fluorescence intensity profiles—human (MM52), M. florum,  BACs
\begin{itemize}
\item Noise considerations (select a set of noise metrics—explain them, justify them and use them consistently throughout the manuscript) that stem from: stretching (show analysis that links stretching with fluorescence signal information content (this curve may look parabolic…..), amount of dye, staing techniques….How fluorescence signals are measured and pre-processed METHODS SECTION (INCA, etc.)
\item Dealing with noise (Introduction)—Systematic vs. random effects. How a signal consensus making approach deals must deal with types of noise through preprocessing (automated ranking of fluorescence signal QUAILTY (mathematical, morphological analysis): uneven stretching, dispersed stretching within an interval dataset, extraneous noise (“fluorescent garbage,” nearby molecules , crossing molecules, etc.). 
\item The Consensus algorithm- Outline algorithm, describe in detail, pseudocode???……..examples of synthetic data, data comparisons using previously. Explain how variants of this algorithm/analysis have been used in other applications…
\item Preprocessing to improve consensus signal making— Describe both mathematical and computational algorithms and workflow. Comparison and analysis of examples drawn from human, synthetic, M. florum, and BAC data, showing how more confident fluoroscans emerge from considered filtering. Use noise metrics to prove this point.
\end{itemize}


