\chapter{Data Structure: Fluorescence Intensity Profiles} \label{Ch2}

Nanocoding \cite{Jo_etal_2007_PNAS}, \cite{Jo_etal_2009}, invented in LMCG, is a more advanced DNA barcoding technique, to obtain genome-wide restriction maps. In nanocoding, molecules are labeled with the restriction enzyme Nb.BbvCI, which cleaves the sequence GC\^\ TGAGG on single strands of double-stranded DNA molecules. The cleaved sites are made detectable by nick translation using fluorochrome-labelled nucleotides. The cleaved sites are called punctates. The labeling is done in test tubes and then presented on nanoscale tubes (fig \ref{fig:Fig2_NanoSlit}) for imaging and subsequent measurements. 
\begin{wrapfigure}{l}{0.5\textwidth}
%\includegraphics[width=0.9\linewidth]{Images/Image_Nanoslit.jpg} 
\begin{center}
\includegraphics[width=0.4\textwidth, bb=0 0 700 350]{Images/Image_Nanoslit.pdf}
\end{center}
\caption{Nanoslits where DNA molecules are presented}
\label{fig:Fig2_NanoSlit}
\end{wrapfigure}
The nanoslits are 1000 nm wide, 100 nm deep and 10 $\mu$m apart. DNA molecules are inherently negatively charged. Hence, they electrokinetically move through the microchannels as relaxed coils until approaching a nanoslit entrance. Then, as one of a molecule enters the nanoslit, it transiently elongates. This is aided by the low-ionic-strength buffers containing the DNA molecules. The DNA molecules are then deposited onto positively charged glass surface. The surface with DNA are then stained with YOYO-I dye, so they can be imaged by fluorescence microscopy. An argon ion laser-illuminated inverted Zeiss 135M microscope was used to image the DNA molecules. An integrated image acquisition and machine vision system was developed in LMCG to automatically detect and analyze molecule within the collected image data. Details of the image processing steps can be found in \cite{Dimalanta_etal_2004_AnalChem}, \cite{Jo_etal_2007_PNAS}, \cite{Ravindran_Gupta_2015_GigaScience}. 

The image processing softwares scans through images of surfaces presented with DNA molecules, as shown below in fig \ref{fig:Fig2_FrameImage}. It shows images of multiple straight DNA molecules which are stained with the YOYO-I dye which fluoresces when excited by the laser-illuminated microscope.  
\begin{figure}[h!]
\begin{center}
%\includegraphics[width=0.55\textwidth, bb=0 0 700 350]{Images/Image4_FrameImage.png}
\includegraphics[scale = 0.45, bb=0 0 1000 600]{Images/Image4_FrameImage.pdf}
\end{center}
\caption{Image of a surface with stained DNA molecules}
\label{fig:Fig2_FrameImage}
\end{figure}

The labelled molecules are imaged by microscopes equipped with two CCD cameras, that have two filtering optics for green and red colors. The green channel acquires images of DNA backbone stained with YOYO-1, and the red channel acquires images of sequence-specific decorations of Alexa Fluor 547 punctuates via fluorescence resonance energy transfer (7). The restriction maps (named Nmaps, for ``nanocoded'' maps) are obtained by meticulous image processing software that analyze the images from the green and red channels, to measure the lengths between two consecutive punctates. 

\begin{figure}[H]
\begin{center}
%\includegraphics[scale=1, bb=0 0 300 150]{Images/Image5_NMap.png}
\includegraphics[scale=1]{Images/Image5_NMap.png}
\end{center}
\caption{5 DNA molecules aligned to reference}
\label{fig:Fig2_NMap}
\end{figure}

These Nmaps are used to align the molecules to an {\emph{in silico}} reference map of a reference genome. Fig \ref{fig:Fig2_NMap} shows how five DNA molecules are nanocoded and aligned to a reference genome. \\

The YOYO-I dye, used to stain the DNA molecules, belongs to the family of monomethine cyanine dyes. Cyanine dyes have a rich history of applications in the photographic industry. Cationic cyanine dyes exhibit very large degrees of fluorescence enhancement on binding to nucleic acids \cite{Rye_etal_1992_NAR}, \cite{Lee_etal_1986_Cytometry}. These characteristics of fluorescence enhancement and high binding affinity are crucial for nanocoding. Netzel, et al, 1995 \cite{Netzel_etal_1995_JPC} observed that there are differences in emission quantum yield between dyes with pyridinium and quinolinium structural components, such as YOYO-1, when bound to $(\text{dAdT})_{10}$ and $(\text{dGdC})_6$ duplexes. In fact, they observed a 2-fold quantum yield increase when switching from AT-rich regions to GC-rich regions. Larsson, et al, 1994, (\cite{Larsson_etal_1994_JACS}) also observed that the fluorescence intensity of YO depends on the base sequence. In fact, they suggested that the quantum yield and fluorescence lifetime for YO complexed with GC-rich DNA sequences are about twice as large as for YO complexed with AT-rich sequences. This is the foundation behind the discovery of {\bf{fluoroscanning: The system of infering about genomic sequence compositions from the observed fluorescence intensity profiles of DNA molecules.}} 

\section{Data description} \label{Ch2_data}
Throughout this thesis, we will use images collected during nanocoding {\emph{Mesoplasma florum (M. florum)}} genome and a multiple myeloma (MM) genome. Multiple myeloma is the malignancy of B lymphocytes
that terminally differentiate into long-lived, antibody-producing plasma cells. {\emph{M. florums}} are members of the class Mollicutes, a large group of bacteria that lack a cell wall and have a characteristically low GC content (\cite{Razin_etal_1998_MMBR}). These diverse organisms are parasites in a wide range of hosts, including humans, animals, insects, plants, and cells grown in tissue culture (\cite{Razin_etal_1998_MMBR}). Aside from their role as potential pathogens, {\emph{M. florums}} are of interest because of their extremely small genome size. The {\emph{M. florum}} has only one chromosome, and its NMap consists of 39 intervals. This implies, the reference genome has 38 restriction sites. The whole genome is approximately 793kb long. The intervals are between 2.111kb and 81.621kb. One of the goals of fluoroscanning it to analyze how similar the fluorescence intensity profiles of segments of different molecules aligned to the same location on a genome are. 
\begin{figure}[H]
\begin{center}
\includegraphics[scale=0.42,page=1]{Plots/MF_Frag15.pdf}
\includegraphics[scale=0.42,page=2]{Plots/MF_Frag15.pdf}
\end{center}
\caption{NMaps aligned to {\emph{M. florum}} Interval 15}
\label{fig:Fig2_MF_Frag15}
\end{figure}

Fig \ref{fig:Fig2_MF_Frag15} shows fluorescence intensity profiles of fragments of 12 DNA molecules that have been aligned to interval 15 of {\emph{M. florum}} genome. The interval is 11.119kb long and each pixel of the captured images correspond to around 209 base pairs on the genome. So, under perfect experimental conditions, each of these fragments should be 53 ($\frac{11119}{209} = 53.2$) pixels long. However, due to several reasons the molecule lengths do not perfectly align with that of the reference. As the dye molecules intercalate between the bases of the DNA molecules, there could be local deformation of the molecules, resulting in more dye molecules to seep in. This could result in slight elongation of the DNA molecule and hence it captures more pixels in the image. This phenomenon of ``stretching'' of molecules is hard to control experimentally, as it involves meticulous measurement of dye and DNA molecule concentrations. Furthermore, when one end of a negatively charged DNA molecule attaches to the positively charged glass surface and the rest of the molecule elongates in the direction of the analyte flow, it also causes differential stretching. 



