%\documentclass[10pt,dvipsnames,table, handout]{beamer} % To printout the slides without the animations
\documentclass[10pt,dvipsnames,table]{beamer} 
%\usetheme{Luebeck} 
%\usetheme{Madrid} 
%\usetheme{Marburg} 
%\usetheme{Warsaw} 
\usetheme{CambridgeUS}
%\setbeamercolor{structure}{fg=cyan!90!white}
%\setbeamercolor{normal text}{fg=white, bg=black}
\setbeamercolor{block title}{bg=red!80,fg=white}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Input header file 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\input{HeaderfileTexSlides}

\logo{\includegraphics[scale=0.4]{uwlogo_web_sm_fl_wht.png}}
%\logo{\includegraphics[width=\beamer@sidebarwidth,height=\beamer@headheight]{uwlogo_web_sm_fl_wht.png}}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% TITLE PAGE 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\DeclarePairedDelimiter\ceil{\lceil}{\rceil}
\title[Statistics for fluoroscanning]{Statistical methods for analyzing images of fluorochrome stained DNA molecules}
\author{Subhrangshu Nandi}
\institute[Prelim exam]{Preliminary Exam \\
Department of Statistics \\
University of Wisconsin-Madison}
\date{March 1, 2016}

\begin{document}
\setlength{\baselineskip}{16truept}
\setbeamertemplate{logo}{}

\frame{\maketitle}

%%%%%%%%%%%%%% Slide 1 %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Background: Genome sequencing}
Genome sequencing \footnote{http://www.genomenewsnetwork.org/} is figuring out the order of DNA nucleotides, or bases that make up an organism's DNA. The haploid human genome is made up of over {\underline{3.2 billion}} base pairs. A DNA sequence that has been translated from life's {\emph{chemical}} alphabet into our alphabet of written letters might look like this:
\begin{figure}[H]
\includegraphics[scale=0.48]{Images/Image_Sequence_1.jpg}
\end{figure}

\begin{figure}[H]
\includegraphics[scale=0.7]{Images/Image_DNA.jpg} \footnote{http://www.differencebetween.info/difference-between-gene-and-genome}
\end{figure}

\note{}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Limitations of Sequencing Technologies \footnote{\tiny{Wikipedia}} }
\vspace{-0.8cm}
\begin{columns}[t]
\begin{column}{0.6\textwidth}
\begin{itemize} 

\item {\small{\bf{Pac Bio}}}
{\footnotesize{only 87\% accuracy; moderate throughput }}

\item {\small{\bf{Ion Torrent Sequencing}}}
{\footnotesize{up to 400 bp read length; homopolymer errors }}

\item {\small{\bf{Illumina sequencing}}}
{\footnotesize{up to 300 bp read length; very expensive equipment}}

\item {\small{\bf{{Sanger sequencing}}}
{\footnotesize{impractical for large sequencing projects}}

\end{itemize}
\end{column}

\begin{column}{0.4\textwidth}
\begin{center}
\begin{figure}[t]
\includegraphics[scale=0.28]{Images/sequencingtechnology.pdf} \footnote{\tiny{https://www.broadinstitute.org/ \\blog/celebrating-fruits-human-genome-sequence}}
\end{figure}
\end{center}
%{\bf{Pac Bio}} Avg 14 kb read length; \textcolor{red}{only 87\% accuracy; moderate throughput} \\
%{\bf{Ion Torrent Sequencing}} \textcolor{red}{up to 400 bp read length; \$32 mil for 1 genome}
\end{column}

\end{columns}

\begin{block}{Common limitations}
\begin{itemize}
{\small{
\item CANNOT completely sequence a human genome because of repititive structures \footnote{\cite{Lander_etal_2001_Nature}} and other complexities
\item CANNOT detect heterozygotes \footnote{\cite{Wheeler_etal_2008_Nature}}
\item are Inadequate for analyzing complex cancer genomes, or other polygenic diseases
}}
\end{itemize}
\end{block}
\note{}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Motivation: mp3 representation of a Genome}
\vspace{-0.8cm}
\begin{columns}[t]
\begin{column}{0.6\textwidth}
\begin{itemize} {\small{
\item Lower resolution: of the order of 400 - 600 bp
\item Simple to obtain
\item Contains the important structural information about a genome
}}
\end{itemize}
\end{column}

\begin{column}{0.4\textwidth}
\begin{figure}[t]
\includegraphics[scale=0.08]{Images/mp3_wav.jpg} \footnote{\tiny{http://coderzen.blogspot.com/ \\ 2014/03/wav-file-vs-mp3-file.html}}
\end{figure}

\end{column}
\end{columns}
\begin{block}{Motivation}
We want to construct signature fluorescence intensity profile of a genome.
\end{block}

\vspace{-0.2cm}
\begin{figure}[H]
\includegraphics[scale=0.3]{Plots/chr13_frag1000_consensusOnly.pdf}
\hspace{0.5cm}
\includegraphics[scale=0.28]{Plots/chr13_frag1025_consensusOnly.pdf} \\
\includegraphics[scale=0.3]{Plots/chr13_frag1050_consensusOnly.pdf}
\hspace{0.5cm}
\includegraphics[scale=0.3]{Plots/chr13_frag1052_consensusOnly.pdf} \\
\includegraphics[scale=0.28]{Plots/chr13_frag1064_consensusOnly.pdf}
\hspace{0.5cm}
\includegraphics[scale=0.3]{Plots/chr13_frag1088_consensusOnly.pdf}
\end{figure}

\note{}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Data obtained from Nanocoding system}
%\vspace{-0.5cm}
\begin{figure}[T]
\includegraphics[scale=0.5]{Images/Image_Nanocoding.jpg}
\end{figure}

\note{}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Nanocoding - alignment of molecules to reference}
\vspace{-1cm}
\begin{figure}[t]
\includegraphics[scale=0.6]{Images/NMaps_Aligned.png}
\hspace{1cm}
\includegraphics[scale=0.2]{Images/BarcodeImage.jpg}
\end{figure}

{\emph{M. florum}} reference map: \\
{\footnotesize{81.6 18.7 59.4 13.9 9.0 5.0 12.3 10.2 15.0 25.4 3.9 20.9 15.6 10.2 9.5 11.1 4.5 
13.7 26.3 38.3 2.1 31.0 19.1 3.6 32.2 41.3 9.8}} \textcolor{Red}{16.4 15.7 6.3 36.5 18.3} {\footnotesize{32.1 21.0 
3.3 14.3 51.3 16.0 17.9}} \\
\vspace{0.5cm}
Molecule {\small{$2064486\_0\_11$}}: {\bf{31.24 \textcolor{Red}{16.80  15.77  6.25  35.08  18.12}}}

\note{
The sequence of lengths between punctates is called a nanocode, akin to barcode\\
It is unique to molecule, and it can be used to determine which location on the genome the molecule is from\\
Example, below is a reference map from a bacteria genome, and a nanocoded molecule
}

\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Nanocoding Images}
\begin{center}
\begin{figure}[H]
\includegraphics[scale=0.3]{Images/IView_Example2.png}
\end{figure}
\end{center}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Fluoroscanning: Beyond Nanocoding}
\begin{figure}[H]
\includegraphics[scale=0.5]{Images/IView_Example2_Mol1.png} \\
\includegraphics[scale=0.5]{Plots/chr13_frag1058_Curve2.pdf} \\

\includegraphics[scale=0.5]{Images/IView_Example2_Mol2.png} \\
\includegraphics[scale=0.5]{Plots/chr13_frag1058_Curve4.pdf} 
\includegraphics[scale=0.5]{Plots/chr13_frag1058_Curve6.pdf} 
\end{figure}

\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Aims}
\begin{block}{Aim 1}
To establish that usable signals with information about genomic sequences could be extracted from the features of the fluorescence intensity profiles of imaged DNA molecules.
\end{block}
\vspace{1cm}
\begin{block}{Aim 2}
To extract two sets of fluorescence intensity profiles in heterozygous regions we want to test the sensitivity of fluoroscanning, i.e., how little of a difference in sequences is translated to detectable dissimilarity in the fluorescence intensity profiles?
\end{block}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Compare intensity profiles of 2 intervals (part of Aim 1)}
\begin{figure}[t]
\includegraphics[scale=0.34, page=2]{Plots/chr13_frag7465.pdf}
%\hspace{0.25cm}
\includegraphics[scale=0.34, page=2]{Plots/chr13_frag7491.pdf}
\end{figure}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Statistical challenges}
\begin{block}{Statistical challenges}
The primary challenge of is to build statistical tools to enable fluoroscanning data analysis. This involves
\begin{itemize}
\item {\bf{Preprocessing}} the data by understanding the different sources of noise, and appropriately reducing their effect or controlling for their presence
\item {\bf{Estimating}} a consensus fluorescence intensity profile for each genomic interval and establish they are sequence dependent
\item Setting up a {\bf{hypothesis testing}} framework where intensity profiles from different genomic intervals could be tested for similarity
\end{itemize}
\end{block}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Iterated Registration}
\begin{itemize}
\item {\bf{Step 1}} Normalize and smooth using B-splines
\item {\bf{Step 2}} Detect functional outliers in the data set, using Fraiman and Muniz functional depth
\item {\bf{Step 3}} Template estimation [{\emph{Expectation step}}]: To estimate the template to register the curves to, we employ a 2-step approach. 
\begin{enumerate}
\item {\emph{Median}}: Estimate $L_1\text{-Median}$, where $L_1\text{-Median }$ $y_m$ is the minimizer of 
\[ \sum\limits_{i = 1}^n \|y_i - y_m \| \]
where $y_i \in \Real^p, \ i = 1,\dots,n$ and $\|u \| = \sqrt{\sum\limits_{j = 1}^p u_j^2}$. Here we ensure that the median is estimated only from the curves not deemed as ``functional outliers'' in Step 2. 
\item {\emph{Weighted mean}}: Estimate the similarity indices $\rho(y_i, y_m), \ i = 1,\dots,n$ and estimate the template $\phi$ as the weighted average of the curves, 
\[ \phi = \frac{1}{n}\sum\limits_{i = 1}^n \rho(y_i, y_m) y_i \]
\end{enumerate}
\end{itemize}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Iterated Registration (contd.)}
\begin{itemize}
\item {\bf{Step 4}} Registration with varying roughness penalty [{\emph{Maximization step}}]: We use the minimum second eigenvalue method to register the curves to the template $\phi$. 
\item {\bf{Step 5}} Convergence of iteration: The objective of iterated registration is to maximize the average similarity to the consensus, i.e., maximize 
\[ \bar{\rho}_{\phi, n} = \frac{1}{n} \sum \limits_{i = 1}^{n} \rho(\phi, f_i)\]
Hence, we iterate steps 3 and 4, until
\[ |\bar{\rho}_{\phi_{(t+1)}, n}^{\ (t+1)} - \bar{\rho}_{\phi_{(t)}, n}^{ \ (t)} | < \epsilon \]
where $t$ denotes iteration number, for some predetermined $\epsilon$
\end{itemize}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
%\begin{frame}
%\frametitle{After ``iterated'' registration}
%\begin{figure}[t]
%\includegraphics[scale=0.34, page=6]{Plots/chr13_frag7465_registered.pdf}
%\hspace{0.25cm}
%\includegraphics[scale=0.34, page=6]{Plots/chr13_frag7491_registered.pdf}
%\end{figure}
%\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{After registration, with GC composition, Interval 7465}
\begin{figure}[t]
\includegraphics[scale=0.34, page=6]{Plots/chr13_frag7465.pdf}
%\hspace{0.25cm}
\includegraphics[scale=0.34, page=8]{Plots/chr13_frag7465_registered.pdf}
\end{figure}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{After registration, with GC composition, Interval 7465}
\begin{figure}[t]
\includegraphics[scale=0.3, page=6]{Plots/chr13_frag7465.pdf}
\hspace{0.5cm}
\includegraphics[scale=0.3, page=8]{Plots/chr13_frag7465_registered.pdf}
\end{figure}

\vspace{-0.5cm}
\begin{columns}
\begin{column}{0.5\textwidth}
% latex table generated in R 3.2.2 by xtable 1.8-0 package
% Sat Feb 27 16:10:06 2016
\begin{table}[ht]
\centering
{\footnotesize{
\begin{tabular}{rrrr}
  \hline
  & $\hat{\beta}$ & SE & p-value \\ 
  \hline
  (Intercept) & 0.9788 & 0.0102 & 0.0000 \\ 
  GC & 0.0083 & 0.0306 & 0.7866 \\ 
  GC\_Lag1 & 0.0410 & 0.0260 & 0.1192 \\ 
  GC\_Lead1 & 0.0435 & 0.0260 &  0.0992 \\ 
  \hline
\end{tabular}
\caption{$\tiny{R^2 = 0.1205}$}
}}
\end{table}
\end{column}

\begin{column}{0.5\textwidth}
% latex table generated in R 3.2.2 by xtable 1.8-0 package
% Sat Feb 27 16:12:34 2016
\begin{table}[ht]
\centering
{\footnotesize{
\begin{tabular}{rrrr}
  \hline
  & $\hat{\beta}$ & SE & p-value \\ 
  \hline
  (Intercept) & 0.9686 & 0.0178 & 0.0000 \\ 
  GC & 0.0090 & 0.0536 & 0.8665 \\ 
  GC\_Lag1 & 0.0583 & 0.0456 & 0.2048 \\ 
  GC\_Lead1 & 0.0537 & 0.0456 & 0.2425 \\ 
  \hline
\end{tabular}
\caption{$\tiny{R^2 = 0.0707}$}
}}
\end{table}
\end{column}

\end{columns}

\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{After registration, with GC composition, Interval 7491}
\begin{figure}[t]
\includegraphics[scale=0.34, page=6]{Plots/chr13_frag7491.pdf}
%\hspace{0.25cm}
\includegraphics[scale=0.34, page=7]{Plots/chr13_frag7491_registered.pdf}
\end{figure}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{After registration, with GC composition, Interval 7491}
\vspace{-0.25cm}
\begin{figure}[t]
\includegraphics[scale=0.3, page=6]{Plots/chr13_frag7491.pdf}
\hspace{0.5cm}
\includegraphics[scale=0.3, page=7]{Plots/chr13_frag7491_registered.pdf}
\end{figure}
\vspace{-0.5cm}

\begin{columns}
\begin{column}{0.5\textwidth}
% latex table generated in R 3.2.2 by xtable 1.8-0 package
% Sat Feb 27 16:10:06 2016
\begin{table}[ht]
\centering
{\footnotesize{
\begin{tabular}{rrrr}
  \hline
  & $\hat{\beta}$ & SE & p-value \\ 
  \hline
  (Intercept) & 0.9473 & 0.0070 & 0.0000 \\ 
  GC & 0.0471 & 0.0237 & 0.0503 \\ 
  GC\_Lag1 & 0.0738 & 0.0211 & 0.0008 \\ 
  GC\_Lead1 & 0.0471 & 0.0202 & 0.0222 \\ 
  \hline
\end{tabular}
\caption{$\tiny{R^2 = 0.5372}$}
}}
\end{table}

\end{column}

\begin{column}{0.5\textwidth}
% latex table generated in R 3.2.2 by xtable 1.8-0 package
% Sat Feb 27 16:12:34 2016
\begin{table}[ht]
\centering
{\footnotesize{
\begin{tabular}{rrrr}
  \hline
  & $\hat{\beta}$ & SE & p-value \\ 
  \hline
  (Intercept) & 0.9139 & 0.0075 & 0.0000 \\ 
  GC & 0.0762 & 0.0255 & 0.0038 \\ 
  GC\_Lag1 & 0.1105 & 0.0227 & 0.0000 \\ 
  GC\_Lead1 & 0.0675 & 0.0217 & 0.0026 \\ 
  \hline
\end{tabular}
\caption{$\tiny{R^2 = 0.6972}$}
}}
\end{table}

\end{column}
\end{columns}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{For Aim 1}
\begin{block}{Aim 1}
To establish that usable signals with information about genomic sequences could be extracted from the features of the fluorescence intensity profiles of imaged DNA molecules.
\end{block}
\begin{itemize}
\item[\checkmark] Preprocess the curves to detect outlier molecular fragments
\item[\checkmark] Control for the various type of noises: Different stretch; Different magnitudes of intensity values; Noises surrounding the molecule backbone; 
\item[\checkmark] Estimate consensus by registering fluorescence intensity profiles
\item Build a regression model to relate mean intensity profiles with nucleotide sequence characteristics
\item Test the findings on ribosomal repeats on human genome (same sequence)
\item Test the findings on intensities from different sequences
\end{itemize}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{For Aim 2}
\begin{block}{Aim 2}
To extract two sets of fluorescence intensity profiles in heterozygous regions we want to test the sensitivity of fluoroscanning, i.e., how little of a difference in sequences is translated to detectable dissimilarity in the fluorescence intensity profiles?
\end{block}
\begin{itemize}
\item Extend the iterated registration algorithm to simultaneous cluster two (or more) sets of curves
\item Implement the method on intensity profiles from different sequences and test sensitivity
\item Implement the method on regions in the mm52 genome that has been confirmed by ``Optical mapping'' to be heterozygos
\end{itemize}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{For statistical methods}
\begin{itemize}
\item[\checkmark] Select an appropriate registration algorithm for this data set
\item[\checkmark] Select the most appropriate functional outlier detection method that works best for this data set
\item[\checkmark] Select a test statistic to test the difference between two groups of curves
\item Design a simulation that highlights that iterated registration has more power than the following existing registration algorithm
\begin{enumerate}
\item Single iteration minimum eigen value method (Ramsay, et al)
\item Iterated registration with affine warping of abcissa (Sangalli, et al)
\item Registration using Fisher-rao metric (Srivastava, et al)
\end{enumerate}
\item Build a regression model to relate mean intensity profiles with nucleotide sequence characteristics
\item Extend the iterated registration algorithm to simultaneous cluster two sets of curves
\end{itemize}
\end{frame}

\begin{frame}
\frametitle{Questions please}

Thank you very much!

\end{frame}

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
    {\footnotesize
    \bibliographystyle{apalike}
    \bibliography{bibTex_Reference}
    }
\end{frame}


\end{document}

