\section{Introduction}

\subsection{Motivation}
One of the most important components of precision medicine is analysis and use of an individual's genomic information and development of targeted intervention, tailored to the individual. This will entail building systems that will allow researchers and practitioners to extract important information from a genome, in a parsimonious manner. Next generation sequencing (NGS) systems are enhancing molecular biology in the right direction, but show little promise towards sequencing every human being's genome. In addition, NGS still have difficulty inferring repetitive structures and duplications \cite{Lander_etal_2001_Nature}, further complicated by gaps and errors in the reference genome. Pioneered by laboratory of computational genomics (LMCG) at UW-Madison, {\emph{Fluoroscanning}} is a single-molecule-based whole genome analysis system that can glean extract important information out of a genome in a swift, economical manner. 

The aim of this project is to develop statistical methods for fluoroscanning - analysis of intensity profiles measured from imaged fluorochrome stained DNA molecules. \\
{\bf{Aim 1}} is to establish that fluorescence intensity profiles measured at different regions have discernible differences if the underlying sequence compositions are distinct. \\
{\bf{Aim 2}} is to detect two sets of fluorescence intensity profiles in heterozygous genomic loci. \\
{\bf{Aim 3}} is to relate features of fluorescence intensity profiles to nucleotide sequence composition. \\
These analyses will be performed on experiments conducted at LMCG, on both ${\emph{M. florum}}$ and a human genome. 

\subsection{Background}
\subsubsection*{Optical Mapping}
Pioneered by LMCG, single molecule genome mapping techniques like optical mapping \cite{Schwartz_etal_1993_Science}, \cite{Dimalanta_etal_2004_AnalChem}, and nanocoding \cite{Jo_etal_2007_PNAS} have changed the landscape of whole genome analysis. Optical Mapping is a novel platform for analyzing genomes: it uses measurements of single DNA molecules to infer a high-resolution genome-wide restriction map, whose representation of genome structure complements genome sequence to yield biological insight. Information about genomic variation can be obtained from these restriction maps, that do not record the full nucleotide sequence. A physical map is a listing of the locations along the genome where certain markers occur. A restriction map is a physical map induced by restriction enzymes. The ordered sequence of distances in base pairs between successive marker positions summarizes the genome sequence and can be viewed as a sort of bar code of the genome. Genomic differences can affect the presence or absence of markers, the distances between them and their orientation, inducing analogous changes in the bar code. Having a reference copy of the human genome allows us to perform such {\emph{in silico}} experiments. 

\subsubsection*{Nanocoding}
Nanocoding \cite{Jo_etal_2007_PNAS}, \cite{Jo_etal_2009}, also invented in LMCG, is a more advanced DNA barcoding technique, to obtain genome-wide restriction maps. In nanocoding, the restriction enzyme cleaves the single strands of double-stranded DNA molecules. The cleaved sites are made detectable by nick translation using fluorochrome-labelled nucleotides. The cleaved sites are called punctates. The biggest advantage over optical mapping is being able to retain long DNA molecules intact and hence improving labeling efficiency. DNA molecules are stained with an intercalating dye YOYO-1 (Oxazole Yellow - abbreviated YO, hence the name YOYO). The labelled molecules are imaged by microscopes equipped with two CCD cameras. The restriction maps are obtained by meticulous image processing software that analyze the images to measure the lengths between two consecutive punctates. 

\subsubsection*{Dye - DNA interaction} 
The dye that is used to stain the DNA molecules, YOYO, exhibit very large degrees of fluorescence enhancement on binding to nucleic acids \cite{Rye_etal_1992_NAR}, \cite{Lee_etal_1986_Cytometry}. These  characteristics of fluorescence enhancement and  high binding affinity are crucial for nanocoding. Netzel, et al, 1995 \cite{Netzel_etal_1995_JPC} observe a 2-fold quantum yield increase when switching from AT-rich regions to GC-rich regions. Larsson, et al, 1994, (\cite{Larsson_etal_1994_JACS}) also observe that the fluorescence intensity of YO depends on the base sequence. In fact, their observation suggests that the quantum yield and fluorescence lifetime for YO complexed with GC-rich DNA sequences are about twice as large as for YO complexed with AT-rich sequences. 

\subsection{Discovery of a new effect - Fluoroscanning}
In nanocoding, the microscope, equipped with CCD cameras, capture the images of fluorescent DNA molecules, dyed with YOYO-1. From the physical and chemical properties of the differential interaction between the dye molecules and the nucleotides mentioned above, it is clear that the fluorescence intensity profiles of the DNA molecules will be strongly dependent on the underlying nucleotide sequence composition. In fact, DNA molecules from the same region on the genome should have similar fluorescence intensity profiles. In addition, different regions on the genome, with discernible differences between their sequence compositions, should exhibit discernible difference in their corresponding fluorescence intensity profiles. This novel technique of inferring about genomic sequence compositions from analyzing fluorescent intensity profiles of imaged DNA molecules is termed {\bf{Fluoroscanning}}. \\

The aim of this project is to develop statistical methods for fluoroscanning data. 
