\section{Introduction}

\subsection{Motivation}
An important component of precision medicine centers on obtaining and interpreting an individual's genomic background for the development of targeted interventions, tailored to that individual. Much of this effort will center on building large databases using critically developed approaches, both ``wet'' and computational that will allow researchers and clinicians to wield detailed, molecular insights into disease that are specifically tailored to the individual. Although contemporary sequencing approaches have greatly accelerated the pace of discovery in the bio-sciences, current approaches show little promise towards achieving comprehensive analysis of large, human populations. Accordingly, the issues surrounding ``universal genome analysis'' present challenges across a broad range of scientific disciplines and engineering domains.

Our own thinking about the problems of how to build genome analysis systems, appropriate for large human populations, are centered on parsimonious considerations that take into account the structure of genomes across human populations and the intrinsic advantages that single molecule approaches provide for the creation and interpretation of very large data-sets.

Pioneered by the Laboratory of Computational Genomics (LMCG), Fluoroscanning is a single-molecule-based system conceived for whole genome analysis that will provide information that is complementary to DNA sequencing and in many applications, we believe it may supplant it. The basic principles of Fluoroscanning are simple: very long DNA molecules (as long as 1,000 microns) are stained with a fluorochrome (a dye molecule that fluoresces upon excitation with light) and imaged by fluorescence microscopy. Such molecules look like long strings, as shown in Fig () and the amount of
fluorescence (light given-off by the stained molecule) measured along its backbone varies. Our hypothesis is that this variation in measured fluorescence varies according to the DNA sequence imaged at each point along a DNA molecule. Because light microscopy has a spatial resolution of about 200 nm, we do not enjoy single base resolution. Instead, fluorescence measurements locally average about 600 bp of DNA, which are collectively based on the optical characteristics of the imaging system (objective, lenses, cameras, dyes, etc.) and how DNA molecules are presented for imaging. Lastly, a slew of automation approaches (workflows) enable creation of large data-sets and analyses comprising individually analyzed DNA molecules.

%The aims of my project are to develop statistical methods for Fluoroscanning, predicated on the analysis of fluorescence intensity profiles of single DNA molecules that reside within large machine vision datasets.\\
%\noindent
%{\bf{Aim 1}} is to establish that fluorescence intensity profiles vary, in a measurable way, according to the composition of the underlying sequence. \\
%\noindent
%{\bf{Aim 2}} is to confidently infer detect two sets of fluorescence intensity profiles that will report the presence of genomic heterozygosity (two genotypes).

\subsection{Background}
\subsubsection*{Optical Mapping}
Pioneered by LMCG, single molecule genome analysis system like optical mapping \cite{Schwartz_etal_1993_Science}, \cite{Dimalanta_etal_2004_AnalChem}, \cite{Aston_etal_1999_Optical}, \cite{Teague_etal_2010_PNAS}, \cite{Yokota_etal_1997_NAR} and nanocoding \cite{Jo_etal_2007_PNAS}, \cite{Chen_etal_2005_Macromolecules} have enabled comprehensive genome analysis in ways that
complement DNA sequencing. Optical Mapping constructs an ordered restriction endonuclease map (physical map) that completely spans across a genome. In practice, the Optical Mapping system combines microfluidic devices, automated imaging and computational workflows that create large data-sets comprising restriction maps derived from millions of cleaved DNA molecules. Reliable genomic markers are created by the restriction enzyme (``molecular scissors'', cutting at specific sequences 4 - 8 bp) via cleavage of a DNA molecule which is imaged by fluorescence microscopy as gap. An ordered restriction map simply tabulates all of the distances between such markers. Such distance measurements provide a large compendium of genomic rulers that detect many forms of polymorphisms and mutations which alter spacing, obliteration, inclusion of markers, and rearrangements. For example, the human genome has $~320,000$ BamHI (a restriction enzyme) sites. Importantly, through the comprehensive analysis of these maps (\cite{Gupta_etal_2015_PNAS}, \cite{Ray_etal_2013_BMCGenomics}, \cite{Teague_etal_2010_PNAS}), genomic structural variation (polymorphism, mutations $>$ 1 kb) is revealed in ways that evade sequencing approaches. Fig \ref{fig:OpticalMapping} shows the workflow of the the Optical Mapping system.

The ordered restriction map of a single DNA molecule (Rmap) is akin to a single sequence read and like sequence data, statistical and computational methods have been developed for alignment, {\emph{de novo}} genome assembly and detection of structural variants using large Rmap data-sets. \cite{Valouev_etal_2006_JCB}, \cite{Valouev_etal_2006_PNAS}, \cite{Valouev_etal_2006_BioInfo}, \cite{Sarkar_etal_2012_JCB}. Having a reference copy of the human genome allows us to perform such in silico experiments. An important consideration here is that {\emph{in silico}} restriction maps (restriction map created in the computer) are commonly developed from the sequence of reference genomes, such as human and this resource is used for alignment and functional annotation of optical maps.

\subsubsection*{Nanocoding}
Nanocoding \cite{Jo_etal_2007_PNAS}, \cite{Jo_etal_2009}, also invented in LMCG, is a more advanced DNA barcoding technique, to obtain genome-wide restriction maps. In place of markers being gaps resulting from restriction digestion, restriction sites are directly fluorochrome labeled - no gaps. In Nanocoding, the restriction enzyme cleaves only one strand (termed ``nicking'') of double-stranded DNA molecules. The cleaved sites are made detectable by covalent incorporation of fluorochrome-labeled nucleotides, which are imaged as small red dots, against the green DNA backbone (stained with YOYO-1, an intercalating fluorochrome). Fig \ref{fig:Nanocoding} illustrates the nanocoding workflow. 

\begin{figure}[H]
\begin{center}
\includegraphics[scale = 0.75]{Images/Nanocoding.png}
\end{center}
\caption{Nanocoding Workflow}
\label{fig:Nanocoding}
\end{figure}

Other advances makes use of nanoconfinement approaches, which load DNA molecules in to nanoscale slits for presentation. Another version of Nanocoding utilizes the same microfluidic devices developed
for Optical Mapping to present surface-mounted molecules for analysis. All of my analyses presented here used machine vision data-sets created from this version of Nanocoding.

\subsubsection*{Dye - DNA interaction} 
The dye that is used to stain the DNA molecules, YOYO, exhibit very large degrees of fluorescence enhancement on binding to nucleic acids \cite{Rye_etal_1992_NAR}, \cite{Lee_etal_1986_Cytometry}. These  characteristics of fluorescence enhancement and  high binding affinity are crucial for nanocoding. Netzel, et al, 1995 \cite{Netzel_etal_1995_JPC} observe a 2-fold quantum yield increase when switching from AT-rich regions to GC-rich regions. Larsson, et al, 1994, (\cite{Larsson_etal_1994_JACS}) also observe that the fluorescence intensity of YO depends on the base sequence. In fact, their observation suggests that the quantum yield and fluorescence lifetime for YO complexed with GC-rich DNA sequences are about twice as large as for YO complexed with AT-rich sequences. 

\subsection{Discovery of a new effect - Fluoroscanning} \label{Aims}
In Nanocoding, the fluorescence microscopy imaging system automatically acquires and analyzes images of fluorescent DNA molecules, dyed with YOYO-1. From differential interaction between the dye molecules and the nucleotides mentioned above, it is clear that the fluorescence intensity profiles of the DNA molecules will be strongly dependent on the underlying nucleotide sequence composition. This novel technique of inferring about genomic sequence compositions from analyzing fluorescent intensity profiles of imaged DNA molecules is termed {\bf{Fluoroscanning}}. 

To summarize, the aims of my project are to develop statistical methods for Fluoroscanning, predicated on the analysis of fluorescence intensity profiles of single DNA molecules that reside within large machine vision data-sets.\\
\noindent
{\bf{Aim 1}} is to establish that fluorescence intensity profiles vary, in a measurable way, according to the composition of the underlying sequence. \\
\noindent
{\bf{Aim 2}} is to detect two sets of fluorescence intensity profiles that will report the presence of genomic heterozygosity (two genotypes).



