\section{Data Structure} \label{Ch2}

\subsection{Fluorescence Intensity Profiles}
An integrated image acquisition and machine vision system was developed in LMCG to automatically detect and analyze molecule within the collected image data. Details of the image processing steps can be found in \cite{Dimalanta_etal_2004_AnalChem}, \cite{Jo_etal_2007_PNAS}, \cite{Ravindran_Gupta_2015_GigaScience}. INCA (not published) is machine vision software developed by Dr. Prabu Ravindran that considers two color, Nanocoding image data-sets and extracts restriction maps from individual molecules to create larges files of Nmaps (1 Nmap = one restriction mapped DNA molecule). Fig \ref{fig:Fig2_FrameImage} below shows images of multiple straight DNA molecules which are stained with the YOYO-I dye which fluoresce when excited by the laser-illuminated microscope.

\begin{figure}[H]
\begin{center}
%\includegraphics[width=0.55\textwidth, bb=0 0 700 350]{Images/Image4_FrameImage.png}
\includegraphics[scale = 0.45]{Images/Image4_FrameImage.pdf}
\end{center}
\caption{Image of a surface with stained DNA molecules}
\label{fig:Fig2_FrameImage}
\end{figure}

The restriction maps (named Nmaps, for ``nanocoded'' maps) are obtained by meticulous image processing software that analyze the images, to measure the lengths between two consecutive punctuates. 
\begin{figure}[H]
\begin{center}
\includegraphics[scale=0.8]{Images/Image5_NMap.png}
\end{center}
\caption{5 DNA molecules aligned to reference \cite{Kounovsky_etal_2013_Macromolecules}}
\label{fig:Fig2_NMap}
\end{figure}

These Nmaps are used to align the molecules to an {\emph{in silico}} reference map of a reference genome. Fig \ref{fig:Fig2_NMap} shows how five DNA molecules are nanocoded and aligned to a reference genome. Notice that intervals 5 and 6 (from the left) of the reference genome have a depth of 5, i.e., 5 molecular fragments are aligned to that interval. However, interval 4 has a depth of 4. \\

\subsection{Data description} \label{Ch2_data}
Throughout this thesis, we will use images collected during nanocoding {\emph{Mesoplasma florum (M. florum)}} genome and a multiple myeloma (MM) genome. Below are brief data descriptions of the two data-sets.
\subsubsection*{\emph{Mesoplasma florum (M. florum)}}
{\emph{M. florums}} are members of the class Mollicutes, a large group of bacteria that lack a cell wall and have a characteristically low GC content (\cite{Razin_etal_1998_MMBR}). These diverse organisms are parasites in a wide range of hosts, including humans, animals, insects, plants, and cells grown in tissue culture (\cite{Razin_etal_1998_MMBR}). Aside from their role as potential pathogens, {\emph{M. florums}} are of interest because of their extremely small genome size. The {\emph{M. florum's}} NMap consists of 39 intervals. This implies, the reference genome has 38 restriction sites. The whole genome is approximately 793 kb long. The intervals are between 2.111 kb and 81.621 kb. 

One natural consequence of fluoroscanning is that the fluorescence intensity profiles of molecular fragments aligned to the same reference interval will have similar features, since the underlying sequence composition is the same. This will be tested as part of Aim 1 \ref{Aims}. 

\begin{figure}[H]
\begin{center}
\includegraphics[scale=0.42,page=1]{Plots/MF_Frag15.pdf}
\includegraphics[scale=0.42,page=2]{Plots/MF_Frag15.pdf}
\end{center}
\caption{NMaps aligned to {\emph{M. florum}} Interval 15}
\label{fig:Fig2_MF_Frag15}
\end{figure}

Fig \ref{fig:Fig2_MF_Frag15} shows fluorescence intensity profiles of fragments of 12 DNA molecules that have been aligned to interval 15 of {\emph{M. florum}} genome. The interval is 11.119kb long and each pixel of the captured images correspond to around 209 base pairs on the genome. So, under perfect experimental conditions, each of these fragments should be 53 ($\frac{11119}{209} = 53.2$) pixels long. However, due to several reasons the molecule lengths do not perfectly align with that of the reference. As the dye molecules intercalate between the bases of the DNA molecules, there could be local deformation of the molecules, resulting in more dye molecules to seep in. This could result in slight elongation of the DNA molecule and hence it captures more pixels in the image. This phenomenon of ``stretching'' of molecules is hard to control experimentally, as it involves meticulous measurement of dye and DNA molecule concentrations. Furthermore, when one end of a negatively charged DNA molecule attaches to the positively charged glass surface and the rest of the molecule elongates in the direction of the analyte flow, it also causes differential stretching. 

In the {\emph{M. florum}} data-set, we have molecular fragments aligned to 39 intervals on the reference genome. The lengths of these intervals vary from 2.111 kb - 81.620 kb, with the average lengths of these intervals being 20.34 kb. 
% latex table generated in R 3.2.2 by xtable 1.8-0 package
% Tue Feb 16 15:11:00 2016
\begin{table}[H]
\centering
%\begin{tabular}{l*{7}{c}}
\begin{tabular}{lrrr|rrr}
  \hline
  \hline
  \multicolumn{4}{c}{Reference Interval} & \multicolumn{3}{c}{Molecular Fragment lengths} \\
  \hline
   Int  & molecules & pixels & length(kb) & min (kb) & avg (kb) & max (kb)\\ 
  \hline
  \hline
    0 &  66 & 391 & 81.62 & 65.67 & 81.07 & 92.79 \\ 
    1 & 208 & 89 & 18.68 & 13.27 & 18.64 & 21.55 \\ 
    2 & 467 & 284 & 59.40 & 43.92 & 59.24 & 69.39 \\ 
    3 & 734 & 67 & 13.94 & 9.59 & 13.86 & 17.34 \\ 
    4 & 895 & 43 & 9.03 & 6.47 & 8.99 & 11.48 \\ 
    5 & 849 & 24 & 5.04 & 2.14 & 5.02 & 5.90 \\ 
    6 & 939 & 59 & 12.34 & 6.58 & 12.29 & 15.55 \\ 
    7 & 1200 & 49 & 10.24 & 6.74 & 10.20 & 12.22 \\ 
  \hline
  \hline
\end{tabular}
\caption{Coverage of {\emph{M. florum}} data}
\label{tab:mftable}
\end{table}

Table \ref{tab:mftable} lists the coverage of the {\emph{M. florum}} data from interval 0 - 7. The rest of the intervals are in \ref{tab:App_mftable}. For example, interval 7 has 1200 molecular fragments aligned to it. Ideally, all of them should be of length 10.24 kb, but in reality the lengths of these fragments lie between 6.74 kb and 12.22 kb, the average being 10.20 kb. 

\subsubsection*{Human genome - multiple myeloma}
We have nanocoded data of a human genome from DNA samples were prepared from purified CD138 plasma cells (MM-S and MM-R sample) and paired cultured stromal cells (normal) from a 58-y-old male MM patient with International Staging System (ISS) Stage IIIb disease. Multiple myeloma is the malignancy of B lymphocytes that terminally differentiate into long-lived, antibody-producing plasma cells. Although it is a cancer genome, substantial portions of it are still identical to the reference human genome. This genome has been comprehensively analyzed to characterize its genome structure and variation by integrating findings from optical mapping with those from DNA sequencing-based genomic analysis \cite{Gupta_etal_2015_PNAS}. While the {\emph{M. florum}} genome only had 39 intervals, the table below (table \ref{tab:mm52intervals}) lists the number of intervals each chromosome of the human genome contains.

\begin{table}[H]
\centering
\begin{tabular}{c | r || c | r}
  \hline
  \hline
  Chromosome & Number of intervals & Chromosome & Number of intervals \\ 
  \hline
  1 & 26069  & 13 & 9916 \\
  2 & 26772  & 14 & 9764 \\
  3 & 21334  & 15 & 9701 \\
  4 & 19406  & 16 & 9106 \\
  5 & 19359  & 17 & 9291 \\
  6 & 18504  & 18 & 8306 \\
  7 & 16797  & 19 & 5763 \\
  8 & 15828  & 20 & 7311 \\
  9 & 13553  & 21 & 3900 \\
  10 & 15053  & 22 & 4326 \\
  11 & 15212  & X & 15953 \\
  12 & 14371  & Y & 2582 \\
  \hline
  \hline
\end{tabular}
\caption{Number of intervals in each human chromosome}
\label{tab:mm52intervals}
\end{table}
Although there are many more intervals in the MM data-set, the coverage is not as deep as that of the {\emph{M. florum}} data-set. Below are some plots of the coverage of 4 of the chromosomes
\begin{figure}[H]
\begin{center}
\includegraphics[scale = 0.46, page = 1]{Plots/mm52_Coverage.pdf}
\includegraphics[scale = 0.46, page = 6]{Plots/mm52_Coverage.pdf}
\end{center}
\caption{Coverage of intervals (at least 6 kb long)}
\label{fig:Coverage_mm1}
\end{figure}

\begin{figure}[H]
\begin{center}
\includegraphics[scale = 0.46, page = 10]{Plots/mm52_Coverage.pdf}
\includegraphics[scale = 0.46, page = 13]{Plots/mm52_Coverage.pdf}
\end{center}
\caption{Coverage of intervals (at least 6 kb long)}
\label{fig:Coverage_mm2}
\end{figure}

Notice in fig \ref{fig:Coverage_mm1} that in chromosome 1, out of 26,069 intervals, only 12,701 are at least 6kb long, and on an average have a coverage of 30 molecules (NMaps) per interval.

\subsection{Sources of noise in the data} \label{sec_noise}
Following are some sources of noise that at least need to be controlled for, if eliminating them is not feasible
\begin{itemize}
\item When the molecules are stretched on the glass surfaces, different portions of the molecules experience different amount of stretch. Consequently, the features of the intensity profiles could get warped. 
\item In regions where the molecules are stretched too much, more dye molecules could intercalate, resulting in a brighter image. The quantum yield of the dye molecules is sequence dependent. However, that dependence could reduce where there is a higher density of dye molecules. 
\item The DNA molecules might not be straight. This could result in inaccurate intensity profiles. There could be other DNA fragments overlapping longer molecules, resulting in local brightness in the image. Some examples of images which would result in noisy observations are shown below
\begin{figure}[H]
\begin{center}
\includegraphics[scale = 0.5]{Images/Outlier_1.png}
\includegraphics[scale = 0.3]{Images/Outlier_2.png} \\
\includegraphics[scale = 0.35]{Images/Outlier_3.png}
\includegraphics[scale = 0.45]{Images/Outlier_4.png}
\end{center}
\caption{Image 1: Molecules too close to each other; \\ Image 2: Too much overlap, and curves molecules; \\Image 3: Fragments overlapping DNA molecules; \\Image 4: Molecules not straight}
\label{fig:Fig2_OutlierImages}
\end{figure}

\end{itemize}

\section{Research questions}
\noindent
{\bf{Aim 1}}: We want to establish that usable signals with information about genomic sequences could be extracted from the features of the fluorescence intensity profiles of imaged DNA molecules. \\
\noindent
{\bf{Aim 2}}: A diploid organism (like human being) is ``heterozygous'' at a genomic locus when there are multiple genotypes at that locus. To extract two sets of fluorescence intensity profiles in heterozygous regions we want to test the sensitivity of fluoroscanning, i.e., how little of a difference in sequences is translated to detectable dissimilarity in the fluorescence intensity profiles? \\

\begin{tcolorbox}[colback=green!5,colframe=green!40!black,title=Statistical challenges] %green!40!black=40%green and 60%black
The primary challenge of is to build statistical tools to enable fluoroscanning data analysis. This involves
\begin{itemize}
\item Preprocessing the data by understanding the different sources of noise, and appropriately reducing their effect or controlling for their presence
\item Estimating a consensus fluorescence intensity profile for each genomic interval and establish they are sequence dependent
\item Setting up a hypothesis testing framework where intensity profiles from different genomic intervals could be tested for similarity
\end{itemize}
\end{tcolorbox}


